src/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -249,49 +298,51 @@ func (dsInfo *builderInfo) volumes() []corev1.Volume {
 }
 
 func (dsInfo *builderInfo) imagePullSecrets() []corev1.LocalObjectReference {
-	pullSecretName := dsInfo.instance.PullSecret()
-	pullSecrets := make([]corev1.LocalObjectReference, 0)
+	if dsInfo.instance == nil {
+		return []corev1.LocalObjectReference{}
+	}
 
-	pullSecrets = append(pullSecrets, corev1.LocalObjectReference{
-		Name: pullSecretName,
-	})
-	return pullSecrets
+	return []corev1.LocalObjectReference{{Name: dsInfo.instance.PullSecret()}}
 }
 
 func (dsInfo *builderInfo) securityContext() *corev1.SecurityContext {
 	var securityContext corev1.SecurityContext
-	if dsInfo.instance.NeedsReadOnlyOneAgents() {
+	if dsInfo.instance != nil && dsInfo.instance.NeedsReadOnlyOneAgents() {
 		securityContext.RunAsNonRoot = address.Of(true)
 		securityContext.RunAsUser = address.Of(int64(1000))
 		securityContext.RunAsGroup = address.Of(int64(1000))
 	}
 
-	if dsInfo.instance.IsOneAgentPrivileged() {
+	if dsInfo.instance != nil && dsInfo.instance.IsOneAgentPrivileged() {
```

Comment:
`dsInfo.instance.IsOneAgentPrivileged()` should always return `true` if classicFullStack is used